libtsuba/include/tsuba/Cache.h

```diff
@@ -0,0 +1,183 @@
+#ifndef KATANA_LIBTSUBA_TSUBA_CACHE_H_
+#define KATANA_LIBTSUBA_TSUBA_CACHE_H_
+
+// This is single threaded only, it is not intended to store large objects,
+// but rather metadata (e.g., a shared_ptr to a property column).
+
+// The problem witchel had implementing a multi-threaded version using
```

Comment:
This comment seems somewhat misplaced as the current implementation is not multithreaded, so why even mention that you were unable to make a multithreaded implementation?

Reply:
It is a "notes for future me/maintainer" sort of thing.  I know in 6 months I'm going to wonder why I didn't just do this as a multi-threaded cache.